processor/signozspanmetricsprocessor/factory.go

```diff
@@ -28,6 +29,8 @@ const (
 	typeStr = "signozspanmetrics"
 	// The stability level of the processor.
 	stability = component.StabilityLevelBeta
+
+	signozID = "signoz.collector.id"
```

Comment:
@srikanthccv will this produce a dotted named key or underscored in final labels? I think we have been following underscore naming in labels till now.

Reply:
This will be converted to _ in the metrics exporter

Reply:
`resource_signoz_collector_id` and `signoz_collector_id` have duplicate values? How is it happening?

Reply:
If I recall correctly, we added a `resource_` prefix for all the attributes to be included so that they could be used for filtering by resource attributes on the services page.

Reply:
I am not sure how this is being consumed in the front end. I can look up to see if this can be removed https://github.com/SigNoz/signoz-otel-collector/blob/b0f42d0f8369d275fd51e83c73d7b1b64d8b2fa9/processor/signozspanmetricsprocessor/processor.go#L863-L871

Reply:
Inherited from

https://github.com/SigNoz/opentelemetry-collector-contrib/blob/c28c2913122ee25c131d29689544d61c9387ed92/processor/signozspanmetricsprocessor/processor.go#L687-L696

Reply:
I looked into frontend implementation. It makes a request to query service to match with `resource_` prefix and then strips away the `resource_` part to make the key look like a trace dot separated trace attribute. This is done just for the span metrics. I don't think I have the complete context behind the reasoning. We could get rid of `signoz_collector_id` for APM metrics from as far as I see, but I think this means we should prefix all resource attributes with `resource_` in metrics exporter which hasn't been the case with `resource_to_telemetry_conversion`.

https://github.com/SigNoz/signoz/blob/b23d8da96c2c21141528af72ea216e6f93cdfd5a/frontend/src/container/MetricsApplication/ResourceAttributesFilter/utils.ts#L24-L34

https://github.com/SigNoz/signoz/blob/b23d8da96c2c21141528af72ea216e6f93cdfd5a/frontend/src/lib/resourceAttributes.ts#L6-L16

Reply:
yeah...you got almost all pieces right. We add prefix to resource attributes and not add a copy. Here I see the exiting `signoz_collector_id` remains to exist.

Reply:
I am not sure if we can get rid of this duplication because there could be mixed use of them by users. I think I will open an issue for this, what do you think?

Reply:
yeah .. okay.. we can discuss them in another issue